docs/en/guides/pipeline-integrations/administration/config-staging-and-rollout.md

```diff
@@ -6,3 +6,489 @@ lang: en
+```shell
+install_location.yml
+pipeline_configuration.yml
+```
+Now git will ignore those two system files. We can go ahead and add the rest of the files in the configuration, commit them and push them to the studio repository.
+```shell
+git add --all
+git commit -am "initial commit of our studio config!"
+git push
+```
+### Creating a new project from git
+
+When you create a new project, simply specify a valid git url when the setup process prompts you to enter the path to the configuration to use. Following our example above, we would enter  `username@someserver.com:/studio_config.git`. As part of the project setup process, Toolkit will clone this repository into the  `config`  folder of your new project configuration. This means that you can later on go into this config folder and run git commands. Note that any cloned pipeline configurations will also clone the git repository and will work seamlessly.
+
+### Making changes to a project
+
+Whenever you have made changes to your primary config, you can simply go to your  `config`  folder and then run  `git diff`,  `git commit`  and  `git push`. This would push your changes to the studio configuration that you cloned your project config from.
+
+### Updating a project to have the latest version
+
+Alternatively, if you have updated your studio level config with some changes and you want to pull those down to your project, just go to your  `config`  folder and run a  `git pull`.  **Important**: Note that once you have done this, make sure you run a  `tank cache_apps`  to ensure that all the app versions that your changed config requires are present in the system!
+
+### Advanced git usage: Branches
+
+The default behavior in toolkit is a simple clone operation when a new project is created. The project will be using the git  `master`  branch. For a more advanced setup, if you for example want to keep certain configuration changes for a project private and don't share them with other projects, you can set up a git branch after the project has been created. You then commit to this branch. The changes you want to push to the studio configuration can be merged with the local master branch for the project and then pushed.
+
+### Advanced git usage: Batch deployment
+
+since toolkit keeps a list of all the different configurations for a {% include product %} site via its pipeline configuration entity, it is possible to iterate over this list and execute a  `git pull`  command for each of the items, effectively running a batch update on all projects. This functionality is not currently part of Toolkit. If you need assistance, don't hesitate to reach out to our technical support for more information.
```

Comment:
```suggestion
Since Toolkit keeps a list of all the different configurations for a {% include product %} site via its pipeline configuration entity, it is possible to iterate over this list and execute a `git pull` command for each of the items, effectively running a batch update on all projects. This functionality is not currently part of Toolkit out of the box, but can be scripted using Python and the {% include product %} Python API.
```